compiler/rustc_resolve/src/late/lifetimes.rs

```diff
@@ -3035,18 +3034,26 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
             }
         };
 
-        let mut err = self.report_missing_lifetime_specifiers(span, lifetime_refs.len());
+        let mut spans: Vec<_> = lifetime_refs.iter().map(|lt| lt.span).collect();
+        spans.sort();
```

Comment:
Why do we have to sort and dedup here?

Reply:
My understanding is that `dedup()` only removes _consecutive_ duplicates, i.e. for full deduplication, we have to sort first.

If your question is why we have to deduplicate `spans` at all: The reason is that there can be multiple lifetimes missing in the same spot, e.g.:
```rust
struct S<'a, 'b> {
    a: &'a i32,
    b: &'b i32,
}

fn foo(a: &i32, b: &i32) -> S {todo!()}
```
```
error[E0106]: missing lifetime specifiers
 --> src/main.rs:6:29
  |
6 | fn foo(a: &i32, b: &i32) -> S {todo!()}
  |           ----     ----     ^ expected 2 lifetime parameters
```
Here, we want a _single_ error message, stating that _two_ lifetime parameters are missing at the indicated location. Therefore, we deduplicate the spans (so that only one error message is constructed) and count how many duplicates there were (so that we can say how many lifetime parameters are expected, e.g. 2 above).